playback/src/player.rs

```diff
@@ -432,7 +486,24 @@ impl PlayerInternal {
         let key = self.session.audio_key().request(track.id, file_id).wait().unwrap();
 
         let encrypted_file = AudioFile::open(&self.session, file_id).wait().unwrap();
-        let audio_file = Subfile::new(AudioDecrypt::new(key, encrypted_file), 0xa7);
+
+        let mut decrypted_file = AudioDecrypt::new(key, encrypted_file);
+
+        let mut normalisation_factor: f32 = 1.0;
+
+        if self.config.normalisation {
+            let normalisation_config = self.parse_normalisation(&mut decrypted_file);
+            normalisation_factor = f32::powf(10.0, (normalisation_config.track_gain_db + self.config.normalisation_pregain) / 20.0);
```

Comment:
I would move this code (and the clipping prevention) into a method of `NormalisationCofig`